### PR TITLE
Run the Signature Checker

### PR DESCRIPTION
**Pull request by mernst**
The pull request also contains a few refactorings, to reduce code duplication, numbrer of warnings, or indentation.

### CODE REVIEW

File: framework/src/main/java/org/checkerframework/framework/stub/ToIndexFileConverter.java [hunk 1]
```diff
@@ -540,7 +549,12 @@ private String getJVML(Type type) {
                 new GenericVisitorAdapter<String, Void>() {
                     @Override
                     public String visit(ClassOrInterfaceType type, Void v) {
-                        String typeName = type.getNameAsString();
+                        @SuppressWarnings(
+                                "signature") // https://tinyurl.com/cfissue/658 for getNameAsString
+                        @FullyQualifiedName String typeName = type.getNameAsString();
+                        @SuppressWarnings(
+                                "signature") // TODO FIXME BUG: resolve requires a @BinaryName, but
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/stub/ToIndexFileConverter.java]: TODO.

File: framework/src/main/java/org/checkerframework/framework/type/AnnotationClassLoader.java [hunk 2]
```diff
@@ -547,8 +560,9 @@ private void loadBundledAnnotationClasses() {
      * @param jar the JarFile containing the annotation class files
      * @return a set of fully qualified class names of the annotations
      */
-    private final Set<String> getBundledAnnotationNamesFromJar(final JarFile jar) {
-        Set<String> annos = new LinkedHashSet<>();
+    @SuppressWarnings("signature") // remove when using reflection-util 0.2.0
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/framework/type/AnnotationClassLoader.java]: It looks like the build is now using 0.2.0.

**Comment by mernst** [comment]
> I started review this, but I noticed several TODO. Should these be fixed before I continue to review? I've made comments on the TODOs I noticed.

@smillst There are some TODOS, but it will take considerable time to address them all.  I'm hoping it's OK to commit it with a few warning suppressions and TODOs, and to fix those later.

File: framework/src/main/java/org/checkerframework/framework/stub/ToIndexFileConverter.java (hunk 1 above)
Comment (smillst) [reply framework/src/main/java/org/checkerframework/framework/stub/ToIndexFileConverter.java]: Is this a bug in the Signature Checker or in ToIndexFileConverter?

File: framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java [hunk 3]
```diff
@@ -1497,8 +1499,11 @@ public Void visitMemberSelect(MemberSelectTree tree, AnnotatedTypeMirror type) {
                 // The field is static and final.
                 Element e = TreeUtils.elementFromTree(tree.getExpression());
                 if (e != null) {
-                    String classname = ElementUtils.getQualifiedClassName(e).toString();
-                    String fieldName = tree.getIdentifier().toString();
+                    @SuppressWarnings("signature") // TODO: check
```
[the remark this answers is not in the document]
Comment (smillst) [reply framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java]: I don't understand the comment.  Check what?

**Comment by mernst** [comment]
@smillst I removed a TODO comment, and I clarified other comments (there seem to be bugs in the Checker Framework that the Signature Checker revealed).

**Pull request merged**